src/objective-c/examples/BUILD

```diff
@@ -15,55 +15,154 @@
 
 
+load("//src/objective-c:grpc_objc_internal_library.bzl", "local_objc_grpc_library")
 load("@build_bazel_rules_apple//apple:ios.bzl", "ios_application")
-load(
-    "@com_github_grpc_grpc//bazel:objc_grpc_library.bzl",
-    "objc_grpc_library",
-)
+load("@build_bazel_rules_apple//apple:tvos.bzl", "tvos_application")
+load("@build_bazel_rules_apple//apple:watchos.bzl", "watchos_application", "watchos_extension")
 
 proto_library(
     name = "messages_proto",
-    srcs = ["BazelBuildSamples/messages.proto"],
-    visibility = ["//visibility:public"],
+    srcs = ["RemoteTestClient/messages.proto"],
+)
+
+proto_library(
+    name = "test_proto",
+    srcs = ["RemoteTestClient/test.proto"],
+    deps = [":messages_proto"],
 )
 
-objc_grpc_library(
+# use objc_grpc_library in bazel:objc_grpc_library.bzl when developing outside the repo
+local_objc_grpc_library(
     name = "test_grpc_objc",
-    srcs = ["BazelBuildSamples/rmt/test.proto"],
+    srcs = ["RemoteTestClient/test.proto"],
     use_well_known_protos = True,
     deps = [
-        "//src/objective-c/examples/BazelBuildSamples/rmt:test_proto",
+        "//src/objective-c/examples:test_proto",
     ],
 )
 
 # Proof that without this works without srcs
-objc_grpc_library(
+local_objc_grpc_library(
     name = "test_objc",
     use_well_known_protos = True,
     deps = [
-        "//src/objective-c/examples/BazelBuildSamples/rmt:test_proto",
-    ]
+        "//src/objective-c/examples:test_proto",
+    ],
 )
 
 objc_library(
-    name = "ios-sample-lib",
-    srcs = glob(["BazelBuildSamples/ios-sample/ios-sample/**/*.m"]),
-    hdrs = glob(["BazelBuildSamples/ios-sample/ios-sample/**/*.h"]),
+    name = "Sample-lib",
+    srcs = glob(["Sample/Sample/**/*.m"]),
+    hdrs = glob(["Sample/Sample/**/*.h"]),
     data = glob([
-        "BazelBuildSamples/ios-sample/ios-sample/Assets.xcassets/**/*",
-        "BazelBuildSamples/ios-sample/ios-sample/Base.lproj/**/*"
+        "Sample/Sample/Base.lproj/**",
+        "Sample/Sample/Images.xcassets/**",
     ]),
-    deps = [
-        ":test_grpc_objc",
-    ]
+    deps = [":test_grpc_objc"],
 )
 
 ios_application(
-    name = "ios-sample",
-    bundle_id = "com.google.ios-sample-objc-bazel",
-    families = ["iphone"],
-    minimum_os_version = "9.0",
-    infoplists = ["BazelBuildSamples/ios-sample/ios-sample/Info.plist"],
+    name = "Sample",
+    bundle_id = "grpc.objc.examples.Sample",
+    minimum_os_version = "8.0",
+    infoplists = ["Sample/Sample/Info.plist"],
+    families = [
+        "iphone",
+        "ipad",
+    ],
+    deps = ["Sample-lib"],
     visibility = ["//visibility:public"],
-    deps = [":ios-sample-lib"],
-)
\ No newline at end of file
+)
+
+objc_library(
+    name = "InterceptorSample-lib",
+    srcs = glob(["InterceptorSample/InterceptorSample/**/*.m"]),
+    hdrs = glob(["InterceptorSample/InterceptorSample/**/*.h"]),
+    data = glob([
+        "InterceptorSample/InterceptorSample/Base.lproj/**",
```

Comment:
How are the `data` section used? Does ios_application recognize the `data` section?

Reply:
For linking stuff like `IBAction`'s to storyboards. Maybe hooking up other assets with the storyboard occurs here as well. I'm not sure about the more complicated use cases.

Reply:
What I meant was, I did not see how the files in `data` section is used at build-time by `objc_library` or `ios_application`. From the [doc](https://docs.bazel.build/versions/master/be/objective-c.html#objc_library) of `objc_library`, it did not say that it is extracting storyboards or assets from `data` section. Can you point me to a doc that said so? Or how did you find out that the storyboards should be placed in `data`?

Reply:
bazel.build contains quite some outdated docs and incomplete docs (at least that's the case for ios related stuff). [rules_apple](github.com/bazelbuild/rules_apple/tree/master/doc) repo's ios-related docs are more updated. [Here](https://github.com/bazelbuild/rules_apple/blob/818e795208ae3ca1cf1501205549d46e6bc88d73/doc/resources.md)'s the one you are looking for.

Reply:
Although `objc_library` is a native rule, they still put that part about feeding assets to `data` in the non-native repository. It does feel a bit weird.